### PR TITLE
Code fixes

### PR DESCRIPTION
**Pull request by manisandro**
- Add some missing `const`s
- Add some missing `inline`s

### CODE REVIEW

File: python/core/auto_generated/qgsrendercontext.sip.in [hunk 1]
```diff
@@ -102,7 +102,7 @@ DPI of 88 will be assumed.
 %End
 
 
-    QPainter *painter();
+    QPainter *painter() const;
```
Comment (nyalldawson) [inline python/core/auto_generated/qgsrendercontext.sip.in]: Where is this one required? It looks to me that a const QgsRenderContext is being passed somewhere, when they should always be passed as non-const QgsRenderContext& only if painter operations will be made.

**Comment by nyalldawson** [comment]
Thanks for upstreaming all these fixes @manisandro !

File: python/core/auto_generated/qgsrendercontext.sip.in (hunk 1 above)
Comment (manisandro) [reply python/core/auto_generated/qgsrendercontext.sip.in]: I have indeed passed a `const QgsRenderContext` in certain places, i.e. [1],  where I just need to query the painter device DPI via `int dpi = context.painter()->device()->logicalDpiX();`, or just to query the context map extent.

[1] https://github.com/kadas-albireo/kadas-albireo2/blob/master/kadas/gui/milx/kadasmilxitem.cpp#L316

Comment (nyalldawson) [reply python/core/auto_generated/qgsrendercontext.sip.in]: In that case I'd add a second

    const QPainter* painter() const; 

getter

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**